packages/platform-core/src/config/config_controller.ts

```diff
@@ -0,0 +1,130 @@
+import os from 'os';
+import path from 'path';
+import fs from 'fs';
+
+/**
+ * Used to interact with config file based on OS.
+ */
+export class ConfigController {
+  dirPath: string;
+  configFilePath: string;
+  _store: Record<string, unknown>;
+
+  /**
+   * Initializes common paths to telemetry configs.
+   */
+  constructor(
+    private readonly projectName = 'amplify',
+    private readonly configFileName = 'config.json'
+  ) {
+    this.dirPath = this.getConfigPath(this.projectName);
+    this.configFilePath = path.join(this.dirPath, this.configFileName);
+  }
+
+  /**
+   * Getter for cached config, retrieves config from disk if not cached already.
+   * If the store is not cached & config file does not exist, it will create a blank one.
+   */
+  private get store(): Record<string, unknown> {
+    if (this._store) {
+      return this._store;
+    }
+    if (fs.existsSync(this.configFilePath)) {
+      this._store = JSON.parse(fs.readFileSync(this.configFilePath).toString());
```

Comment:
can we use async methods here?

Reply:
We should also specify `utf-8` format to `readFile` rather than calling `.toString`